### DISCUSSION TITLE
Migration workflow for a new system

### DISCUSSION THREAD

**Discussion by maku** [SeaQL/sea-orm#543] [Q&A]
I would like to understand the recommended migration workflow for a new system.

From my point of view I would like to have 1 source of truth for migration.

So I see this 2 variants:

- Define my changes in a SQL (up and down) and generate Entities etc.  
- Define my changes in Entities and generate the required sql changes 

As far as I know, none of these variants are possible using sea-orm. Is this correct?

I know about the possibility to generate the entities using sea-orm-cli.
And I tried to define a SQL String in the migration rust file. But it is not capable to handle all statements at once... 
Otherwise I would have to split all changes into individual statements...

In the examples, the migration is explicitly programmed on the basis of the entities. But that doesn't seem particularly elegant to me.
It seems like an unnecessary hassle to me. Is this the recommended way to go?

But I'm missing the "big picture" in this area... What would be the best strategy?

**Comment by tyt2y3** [comment]
See also https://github.com/SeaQL/sea-orm/discussions/513

**Comment by billy1624** [comment]
Hey @maku, sorry for the delay, at the moment user has to make sure the database schemas and SeaORM entities are in sync.

Personally, I prefer to defined the entities in Rust and write migrate to alter the database schemas.
